### PR TITLE
go rewrite - copy script and adjustments for compute

### PR DESCRIPTION
**Pull request by c2thorn**
Example commands for datafusion

1. recreate go_*.yaml files in MMv1 directory with empty descriptons/notes
`bundle exec compiler -v "beta" -e terraform -o "$GOPATH/src/github.com/hashicorp/terraform-provider-google-beta" -p products/datafusion --go-yaml`
2. run description-copy.go to insert exact text from ruby yaml
`go run . --yaml` 
3. run go engine
`go run . --version beta --output $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta`

```release-note:none

```

### CODE REVIEW

File: mmv1/products/compute/HaVpnGateway.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Copyright 2023 Google Inc.
```
Comment (c2thorn) [inline mmv1/products/compute/HaVpnGateway.yaml:1]: This is just renaming the old file `VpnGateway`

File: mmv1/products/compute/InstanceGroup.yaml [hunk 2]
```diff
@@ -50,26 +50,6 @@ parameters:
     description: 'A reference to the zone where the instance group resides.'
     required: true
     custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
-  - !ruby/object:Api::Type::Array
```
Comment (c2thorn) [inline mmv1/products/compute/InstanceGroup.yaml:53]: A completely excluded field, removed to make the description script work easier

File: mmv1/products/compute/TargetVpnGateway.yaml [hunk 3]
```diff
@@ -1,106 +0,0 @@
-# Copyright 2023 Google Inc.
```
Comment (c2thorn) [inline mmv1/products/compute/TargetVpnGateway.yaml:1]: Renamed to `VpnGateway.yaml`

File: mmv1/products/compute/VpnGateway.yaml [hunk 4]
```diff
@@ -12,21 +12,17 @@
 # limitations under the License.
```
Comment (c2thorn) [inline mmv1/products/compute/VpnGateway.yaml:12]: See previous comments, this is really a renaming of files between `HaVpnGateway` and `VpnGateway`

**Review by NickElliot — APPROVED** [review]

**Pull request merged**